src/Swift.Bindings/src/Model/TypeSpec.cs

```diff
@@ -0,0 +1,380 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Text;
+
+namespace BindingsGeneration;
+
+public abstract class TypeSpec
+{
+	protected TypeSpec(TypeSpecKind kind)
+	{
+		Kind = kind;
+		GenericParameters = new List<TypeSpec>();
+		Attributes = new List<TypeSpecAttribute>();
+	}
+
+	public TypeSpecKind Kind { get; private set; }
+	public List<TypeSpec> GenericParameters { get; private set; }
+	public bool ContainsGenericParameters { get { return GenericParameters.Count != 0; } }
+	public List<TypeSpecAttribute> Attributes { get; private set; }
+	public bool HasAttributes { get { return Attributes.Count != 0; } }
+	public bool IsInOut { get; set; }
+	public bool IsAny { get; set; }
+	public virtual bool IsEmptyTuple { get { return false; } }
+	protected abstract string LLToString(bool useFullName);
```

Comment:
Why LLToString is needed?

Reply:
ToString in `TypeSpec` takes care of all the decor around a TypeSpec (`inout`, `any` and generic components). Subclasses only need to concentrate on the bulk of writing. This avoids code duplication.